### PR TITLE
add topologySpreadConstraints

### PR DESCRIPTION
**Pull request by jgregmac**
PR supports feature requested here: #1513 

- add topologySpreadConstraints to the logstash statefulSets
- added supporting values to values.yaml with in-line doc link.
- Added and ran templating test
- Added value to the chart README.

### CODE REVIEW

File: logstash/templates/statefulset.yaml [hunk 1]
```diff
@@ -115,6 +115,15 @@ spec:
         podAffinity:
 {{ toYaml . | indent 10 }}
       {{- end }}
+      {{- if .Values.topologySpreadConstraints.enabled }}
+      topologySpreadConstraints:
+        - maxSkew: {{ .Values.topologySpreadConstraints.maxSkew }}
```
Comment (cdenneen) [inline logstash/templates/statefulset.yaml:120]: @jgregmac this should be a loop to allow for more than one constraint.

**Comment by cdenneen** [comment]
@jmlrt allowing issues and PRs to auto stale/close isn't really helpful.
Please re-open this PR and Issue #1513.

@jgregmac  Possibly need to handle multiple like this:

```yaml
spec:
  topologySpreadConstraints:
    - maxSkew: 1
      topologyKey: "topology.kubernetes.io/zone"
      whenUnsatisfiable: ScheduleAnyway
      labelSelector:
        matchLabels:
          dev: jjones
    - maxSkew: 1
      topologyKey: "kubernetes.io/hostname"
      whenUnsatisfiable: ScheduleAnyway
      labelSelector:
        matchLabels:
          dev: jjones
    - maxSkew: 1
      topologyKey: "karpenter.sh/capacity-type"
      whenUnsatisfiable: ScheduleAnyway
      labelSelector:
        matchLabels:
          dev: jjones
```

**Comment by cdenneen** [comment]
@jgregmac you have to sign the CLA before this can be reviewed by folks at @elastic

**Comment by jgregmac** [comment]
I have signed the CLA multiple times, so I don't know why this PR continues to be ignored.  It is frustrating.
________________________________

**Comment by cdenneen** [comment]
@jgregmac please rebase

**Review by cdenneen — COMMENTED** [review]
LGTM

File: logstash/templates/statefulset.yaml (hunk 1 above)
Comment (jgregmac) [reply logstash/templates/statefulset.yaml:120]: @cdenneen This recommendation has been implemented in my fork, although I do not see that change reflected in the "changed files" view of this PR.  The recent merge from master is reflected in the commit history though, so I am puzzled by that. Perhaps I need to file a new PR?

Comment (cdenneen) [reply logstash/templates/statefulset.yaml:120]: https://github.com/elastic/helm-charts/blob/main/CONTRIBUTING.md#requirements-for-submiting-a-pull-request

Maybe new PR is necessary and looks like bumping Chart is necessary requirement too.

**Comment by theanotherwise** [comment]
@cdenneen 

one year later.. 

https://github.com/elastic/helm-charts/pull/1734

**Pull request closed without merging**